### PR TITLE
Add `dotnet-interactive`

### PR DESCRIPTION
**Pull request by dhirschfeld**
Adds https://github.com/dotnet/interactive

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by dhirschfeld** [comment]
Windows works! :tada:
```powershell
(base) ~\code> mamba create -n dotnet-interactive dotnet dotnet-interactive jupyterlab --use-local --yes
(base) ~\code> conda activate dotnet-interactive
(dotnet-interactive) ~\code> jupyter lab
```

image

**Comment by dhirschfeld** [comment]
This is now ready for review. I fully intend to support linux as it's my eventual deployment target - I  just don't have the bandwidth at present to get that up and running now. I'll tackle that in a separate PR on the feedstock unless someone beats me to it...

**Comment by dhirschfeld** [comment]
ping @conda-forge/staged-recipes!

File: recipes/dotnet-interactive/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+build:
+  skip: true  # [not win]
+  number: 0
+
+requirements:
+  build:
+    - dotnet >=5.0.100
+  run:
+    - dotnet >=5.0.100
```
Comment (chrisburr) [inline recipes/dotnet-interactive/meta.yaml:21]: Is this really just a lower bound or should there also be an upper limit? i.e. is a hypothetical dotnet 6 or dotnet 15 expected to work just fine as well?

Comment (dhirschfeld) [reply recipes/dotnet-interactive/meta.yaml:21]: That's an interesting question! It seems minor versions *should* be compatible:
https://docs.microsoft.com/en-us/dotnet/core/versions/selection#framework-dependent-apps-roll-forward

image

...I'll optimistically pin to <6.0.

**Pull request merged**